doc/neps/nep-0056-array-api-main-namespace.rst

```diff
@@ -0,0 +1,524 @@
+and SciPy's [2]_ experiences and impact (large performance gains in some cases
+- ``LinearDiscriminantAnalysis.fit`` showed ~28x gain with PyTorch on GPU vs.
+NumPy) paint a more complete picture.
+
+For end users who are using NumPy directly, little changes aside from there
+being fewer differences between NumPy and other libraries they may want to use
+as well. This shortens their learning curve and makes it easier to switch
+between NumPy and PyTorch/JAX/CuPy. In addition, they should benefit from
+array-consuming libraries starting to support multiple array libraries, making
+their experience of using a stack of Python packages for scientific computing
+or data science more seamless.
+
+Finally, for authors of other array libraries as well as tools like Numba,
+array API standard support should save them time. The design rules ([3]_), and
+in some cases new APIs like the ``unique_*`` ones, are easier to implement on
+GPU and for JIT compilers as a result of more predictable behavior.
+
+
+Backward compatibility
+----------------------
+
+The changes that have a backwards compatibility impact fall into these
+categories:
+
+1. Raising errors for consistency/strictness in some places where NumPy now
+   allows more flexible behavior,
+2. Dtypes of returned arrays for some element-wise functions and reductions,
+3. Numerical behavior for few tolerance keywords,
+4. Functions moved to ``numpy.linalg`` and supporting stacking/batching.
+
+Raising errors for consistency/strictness includes:
+
+1. Making ``.T`` error for >2 dimensions,
+2. Making ``cross`` error on size-2 vectors (only size-3 vectors are supported),
+3. Making ``solve`` error on ambiguous input (only accept ``x2`` as vector if ``x2.ndim == 1``),
+4. ``outer`` raises rather than flattens on >1-D inputs,
+5. In-place operators are disallowed when the left-hand side would be promoted.
+
+Dtypes of returned arrays for some element-wise functions and reductions
+includes functions where dtypes need to be preserved: ``ceil``, ``floor``, and
+``trunc`` will start returning arrays with the same integer dtypes if the input
+has an integer dtype. It also includes dtype changes: ``sum`` and ``prod``
+always upcast lower-precision floating-point dtypes to ``float64`` when
+``dtype=None`` (this upcasting is already done for inputs with lower-precision
+integer dtypes). 
+
+Changes in numerical behavior include:
+
+- The ``rtol`` default value for ``pinv`` changes from ``1e-15`` to a
+  dtype-dependent default value of ``None``, interpreted as ``max(M, N) *
+  finfo(result_dtype).eps``,
+- The ``tol`` keyword to ``matrix_rank`` changes to ``rtol`` with a different
+  interpretation. In addition, ``matrix_rank`` will no longer support 1-D array
+  input,
+- ``argsort`` and ``sort`` will gain a ``stable`` keyword argument in addition
+  to ``kind``, and the default will become ``stable=True``.
+- The ``ddof`` keyword in ``std`` and ``var`` changes its name to
+  ``correction``.
```

Comment:
Do we actually have concrete plans to deprecate it?  IF yes, say that it will be a slow deprecation, as that is important to judge backcompat impact.
I.e. "we plan to deprecate the `ddof` name eventually".

Reply:
I don't think we have plans to deprecate `ddof`. At least I didn't consider it yet, and I'm not sure it's necessary.